sdk/core/core-lro/src/lroEngine/lroEngine.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PollOperationState } from "../pollOperation";
+import { Poller } from "../poller";
+import { LongRunningOperation, LroEngineOptions, ResumablePollOperationState } from "./models";
+import { GenericPollOperation } from "./operation";
+
+/**
+ * The LRO Engine, a class that performs polling.
+ */
+export class LroEngine<TResult, TState extends PollOperationState<TResult>> extends Poller<
+  TState,
+  TResult
+> {
+  private intervalInMs: number;
+
+  constructor(lro: LongRunningOperation<TResult>, options?: LroEngineOptions) {
+    const { intervalInMs = 2000, resumeFrom } = options || {};
+    const state: TState & ResumablePollOperationState<TResult> = resumeFrom
+      ? JSON.parse(resumeFrom).state
```

Comment:
Can we wrap all of the JSON.parse calls in try-catch blocks? Or to have a utility function that does so in a single place? Ideally we would have a standard error message or a set of messages for the known cases of misshapen JSON values.

Reply:
This is the only place where  `JSON.parse` is called. I added trying logic.